pkg/controller/configmap/configmap_controller.go

```diff
@@ -69,14 +93,29 @@ func add(mgr manager.Manager, sharedInfo *sharedinfo.SharedInfo, r reconcile.Rec
 		return err
 	}
 
+	reconcileConfigMap, ok := r.(*ReconcileConfigMap)
+	if !ok {
+		log.Info("Reconciler Interface to ReconcileConfigMap pointer type cast assertion failed")
```

Comment:
Is there any case this could fail?

Reply:
To my understanding, if passed argument:r is not a valid ReconcileConfigMap object, for example, nil object, or other struct type, which will lead to this type cast failed. I don't think there is other case causing fail.

Reply:
This type cast assertion is compulsory.